chain/chain/src/chain.rs

```diff
@@ -3507,6 +3507,12 @@ impl Chain {
     /// Function to check whether we need to create a new snapshot while processing the current block
     /// Note that this functions is called as a part of block preprocessing, so the head is not updated to current block
     fn should_make_snapshot(&mut self) -> Result<SnapshotAction, Error> {
+        if let StateSnapshotConfig::Disabled =
+            self.runtime_adapter.get_tries().state_snapshot_config()
+        {
+            return Ok(SnapshotAction::None);
+        }
+
```

Comment:
Would it be possible for a node operator to be stuck with a snapshot after switching from enabled to disabled? In other words is there any snapshot cleanup when disabled? 

I'm not suggesting we fix it in this PR, mostly just for my information. We discourage disabling state snapshots anyway.

Reply:
nit: that's a funny way to check if it's disabled, I think a simple comparison would be cleaner:
`if self.runtime_adapter.get_tries().state_snapshot_config() == StateSnapshotConfig::Disabled {`

Reply:
I dare to disagree here. Comparison requires deriving `Eq`. Pattern match is more idiomatic for an enum.

Reply:
Good point, agreed, TIL.